### PR TITLE
Add Geo-Replicated SQL MI autorotation note 

### PR DESCRIPTION
**Pull request by lukecalderon**
In `sql-docs\azure-sql\database\transparent-data-encryption-byok-key-rotation.md`

Added a note to provide clarification when using the same default TDE encryptor across a failover group. This is mentioned in a different page [here](https://learn.microsoft.com/en-us/azure/azure-sql/database/transparent-data-encryption-byok-overview?view=azuresql-mi#geo-replication-considerations-when-configuring-automated-rotation-of-the-tde-protector:~:text=TDE%20with%20automatic%20key%20rotation%20must%20be%20configured%20on%20all%20servers%20if%20you%27re%20configuring%20TDE%20in%20the%20Azure%20portal.); however, the article I've updated contradicts the configuration.

The behaviour has been confirmed by the SQL MI Product Group via a support ticket.

### CODE REVIEW

**Comment by Court72** [comment]
@GithubMirek

Can you review the proposed changes? 
	
**Important**: When the changes are ready for publication, adding a `#sign-off` comment is the best way to signal that the PR is ready for the review team to merge. 
	
#label:"aq-pr-triaged"
@MicrosoftDocs/public-repo-pr-review-team

**Comment by VanMSFT** [comment]
Thanks, @lukecalderon - Can you help reference the support ticket?

@GithubMirek - Please confirm if these changes are correct.

**Comment by lukecalderon** [comment]
> Thanks, @lukecalderon - Can you help reference the support ticket?
> 

Sure - MS Support Ref is `2405230030006073`

**Comment by VanMSFT** [comment]
Following up with Mirek.

File: azure-sql/database/transparent-data-encryption-byok-key-rotation.md [hunk 1]
```diff
@@ -188,6 +188,9 @@ Using the [Azure portal](https://portal.azure.com):
 
 When the key is rotated on the primary server, it's automatically transferred to the secondary server.
 
+> [!NOTE]
+> If the same key vault key on the **primary** server is used as the default TDE protector on the secondary server, then ensure **Auto-rotate key** is enabled for **both** servers. Failure to do so may lead to the auto-rotation workflows entering an error state, and prevent further manual key rotation operations.
```
Comment (VanMSFT) [inline azure-sql/database/transparent-data-encryption-byok-key-rotation.md]:
```suggestion
> If the same key vault key on the primary server is used as the default TDE protector on the secondary server, ensure **Auto-rotate key** is enabled for **both** servers. Failure to do so may lead to the auto-rotation workflows entering an error state and prevent further manual key rotation operations.  
```

**Review by lukecalderon — COMMENTED** [review]
LGTM

**Comment by VanMSFT** [comment]
Hi @lukecalderon - I'm getting contradicting information from the Product Team on this. I'll need to investigate further. Thanks.

**Comment by lukecalderon** [comment]
> Hi @lukecalderon - I'm getting contradicting information from the Product Team on this. I'll need to investigate further. Thanks.

No problem, sums up my experience too. Happy to provide any further info on it if needed.

**Comment by VanMSFT** [comment]
> > Hi @lukecalderon - I'm getting contradicting information from the Product Team on this. I'll need to investigate further. Thanks.
> 
> No problem, sums up my experience too. Happy to provide any further info on it if needed.

Hi @lukecalderon - Looking at the Support case you referenced, it doesn't mention that the issue was due to not having **Auto-rotate key** set on both servers. That may have been what was mentioned to you, but our Product Team stated that the setting isn't needed on both servers. Auto rotation can ben enabled on either the primary or the secondary server, and should still work. I'll need to fix language on our other doc as well to reflect this.

If you have more to add or know of the PG person that stated this, I can help follow-up. Thanks!

**Comment by lukecalderon** [comment]
> Hi @lukecalderon - Looking at the Support case you referenced, it doesn't mention that the issue was due to not having **Auto-rotate key** set on both servers. That may have been what was mentioned to you, but our Product Team stated that the setting isn't needed on both servers. Auto rotation can ben enabled on either the primary or the secondary server, and should still work. I'll need to fix language on our other doc as well to reflect this.
> 
> If you have more to add or know of the PG person that stated this, I can help follow-up. Thanks!

Hi @VanMSFT - I was in direct discussion with the engineer (Abdullah Qtaishat) over Teams, who in turn was in discussion with the Product Group, so this may not have made it into the ticketing system.

In our configuration, we had the primary configured with Default TDE/Auto-Rotation:
image

Whilst on the secondary, it was configured without auto-rotation:
image

They both matched, until the primary rotated. The key existed on the secondary server, but a 'background job' got stuck rotating the key onto the secondary. The PG had to manually cancel the job, before I could manually select the new key on the secondary server. Afterwards, the engineer informed me that the PG had stated if the same key is used on the primary and secondary servers, and is the default TDE protector, then auto-rotation must be enabled on both servers.

**Comment by VanMSFT** [comment]
Thanks for the additional context, @lukecalderon! I'll check with them and see what they'll say.

**Comment by lukecalderon** [comment]
> Thanks for the additional context, @lukecalderon! I'll check with them and see what they'll say.

Hi @VanMSFT, how did you get on with the PG?

**Comment by VanMSFT** [comment]
Sorry for the delay @lukecalderon! I'm following up internally.

**Comment by VanMSFT** [comment]
Doing a confirmation check with the PG.

**Comment by VanMSFT** [comment]
PG confirmed. Approving.

#sign-off

**Comment by rwestMSFT** [comment]
@Jak-MS In future, please use a squash merge if there is more than one commit from this repo.

**Pull request merged**